### PR TITLE
fix: various todos

### PR DESCRIPTION
**Pull request by xenide**

### CODE REVIEW

File: src/GenericFactory.sol [hunk 1]
```diff
@@ -146,7 +146,6 @@ contract GenericFactory is IGenericFactory, Owned {
 
         (address lToken0, address lToken1) = _sortAddresses(aTokenA, aTokenB);
 
-        // TODO: Test that _loadCurve errors for invalid indexes.
```
Comment (xenide) [inline src/GenericFactory.sol:149]: this is taken care of by `testCreatePair_CurveDoesNotExist`

File: src/GenericFactory.sol [hunk 2]
```diff
@@ -79,7 +79,6 @@ contract GenericFactory is IGenericFactory, Owned {
             }
 
             lFreeMem += lSize;
-            // TODO: Do we need to pad to 32 bytes?
```
Comment (xenide) [inline src/GenericFactory.sol:82]: this should be fine as it has been working without problems and from our understanding of constructor bytecode

File: src/curve/constant-product/ConstantProductPair.sol [hunk 3]
```diff
@@ -1,5 +1,4 @@
-// TODO: Can we reduce the nesting by deleting the parent dir?
```
Comment (xenide) [inline src/curve/constant-product/ConstantProductPair.sol:1]: I think the nesting makes it clearer which contract is for which curve. But don't have a super strong opinion of this tho

**Comment by xenide** [comment]
@OliverNChalk can you address the two fallback `TODO`s?

File: src/GenericFactory.sol (hunk 2 above)
Comment (OliverNChalk) [reply src/GenericFactory.sol:82]: agreed, can resolve

File: src/curve/constant-product/ConstantProductPair.sol (hunk 3 above)
Comment (OliverNChalk) [reply src/curve/constant-product/ConstantProductPair.sol:1]: Yeah lets leave as is, this was before we split out StablePair from StableMintBurn

**Comment by OliverNChalk** [comment]
If we convert to fallback we no longer implement ReservoirPair abstract mathoeds so instead de-duped the code. We should have a call at some point to discuss whether it is possible to have StableMintBurn extend ReservoirPair to minimize code dupe.

**Review by OliverNChalk — APPROVED** [review]

**Pull request merged**